### ISSUE TITLE
Kubebuilder scaffolding with pkg/ and internal/

### ISSUE THREAD

**Issue by shaneutt** [Kong/kubernetes-ingress-controller#1965]
### Current Behavior

Currently when we want to use `kubebuilder` to scaffold new APIs, webhooks, or controllers, some manual massaging of files must be done as `kubebuilder` is currently highly opinionated (and not configurable) in its destinations for scaffolded packages.

This has been true since we switched to `kubebuilder` and generally is low severity because we can work around it with a little manual work, but a fix would be a nice to have as it's cumbersome and _not intuitive for new contributors_.

### Expected Behavior

Commands to scaffold with `kubebuilder` should generate immediately working and linkable packages without needing to `mv` and `sed` files.

### Anything else?

This issue is effectively a reference for https://github.com/kubernetes-sigs/kubebuilder/issues/2393

**Comment by shaneutt** [comment]
This is blocked on https://github.com/kubernetes-sigs/kubebuilder/issues/932

**Comment by mflendrich** [comment]
Closing - we'll deal with the inconvenience for the time being and meanwhile @shaneutt is working upstream on getting the importance of this issue noticed.
